include/libtorrent/utp_socket_manager.hpp

```diff
@@ -119,6 +120,9 @@ namespace libtorrent
 		// the counter is the enum from ``counters``.
 		void inc_stats_counter(int counter, int delta = 1);
 
+		packet *acquire_packet(std::uint16_t allocate) { return m_packet_pool.acquire(allocate); }
```

Comment:
I don't think ``allocate`` should be ``uint16_t``. I think it should be ``int`` because it will be an arithmetic type. you can have a ``TORRENT_ASSERT`` internally to ensure it fits in a ``uint16_t``